### PR TITLE
Introduce deleteRow() method for TemplateProcessor

### PR DESCRIPTION
**Pull request by simivar**
### Description

#1017 done again after 4 years

### CODE REVIEW

**Comment by dodubassman** [comment]
This needs to be merged 🙏

**Comment by n-osennij** [comment]
+1

**Comment by Anubarak** [comment]
Why isn't this merged yet? I really need this but I don't want to create a custom fork just for this feature

**Comment by simivar** [comment]
Maybe I'll ping @Progi1984 @PowerKiKi as it has 5 years now! :D

File: src/PhpWord/TemplateProcessor.php [hunk 1]
```diff
@@ -766,6 +766,77 @@ public function cloneRow($search, $numberOfClones)
...
+    /**
+     * Delete a table row in a template document.
+     *
+     * @param string $search
+     *
+     * @return void
+     *
+     * @throws \PhpOffice\PhpWord\Exception\Exception
+     */
+    public function deleteRow($search)
```
Comment (Progi1984) [inline src/PhpWord/TemplateProcessor.php:778]:
```suggestion
    public function deleteRow(string $search): void
```

File: src/PhpWord/TemplateProcessor.php [hunk 2]
```diff
@@ -1088,6 +1159,43 @@ protected function getDocumentContentTypesName()
...
+    /**
+     * Find the start position of the nearest table before $offset.
+     *
+     * @param integer $offset
+     *
+     * @return integer
+     *
+     * @throws \PhpOffice\PhpWord\Exception\Exception
+     */
+    protected function findTableStart($offset)
```
Comment (Progi1984) [inline src/PhpWord/TemplateProcessor.php:1171]:
```suggestion
    protected function findTableStart(int $offset): int
```

File: src/PhpWord/TemplateProcessor.php [hunk 3]
```diff
@@ -1088,6 +1159,43 @@ protected function getDocumentContentTypesName()
...
+    }
+
+    /**
+    * Find the end position of the nearest table row after $offset.
+    *
+    * @param integer $offset
+    *
+    * @return integer
+    */
+    protected function findTableEnd($offset)
```
Comment (Progi1984) [inline src/PhpWord/TemplateProcessor.php:1194]:
```suggestion
    protected function findTableEnd(int $offset): int
```

File: tests/PhpWord/TemplateProcessorTest.php [hunk 4]
```diff
@@ -179,6 +179,33 @@ final public function testXslStyleSheetCanNotBeAppliedOnFailureOfLoadingXmlFromT
         @$templateProcessor->applyXslStyleSheet($xslDomDocument);
     }
 
+    /**
+     * @covers ::getVariables
+     * @covers ::deleteRow
+     * @covers ::saveAs
+     * @test
+     */
+    public function testDeleteRow()
```
Comment (Progi1984) [inline tests/PhpWord/TemplateProcessorTest.php:188]:
```suggestion
    public function testDeleteRow(): void
```

**Review by Progi1984 — COMMENTED** [review]
@simivar Thanks for the ping.

Could you rebase your PR ? and fix my feedback ? Thanks in advance

**Comment by simivar** [comment]
@Progi1984 done

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
@simivar Could you fix PHPCSFixer, please ?

**Comment by simivar** [comment]
@Progi1984 done

**Comment by simivar** [comment]
@Progi1984 kindly pinging 🙏

**Pull request merged**